pkg/kube/permissions.go

```diff
@@ -0,0 +1,62 @@
+package kube
+
+import (
+	"bytes"
+	"fmt"
+	"os/exec"
+	"strings"
+)
+
+// Resource is the representation of any Kubernetes resource
+type Resource string
+
+// Verb is the representation of the different verbs that can be checked for Kubernetes resources
+type Verb string
+
+const (
+	// ClusterRoleBindings is the clusterrolebindings.rbac.authorization.k8s.io resource
+	ClusterRoleBindings Resource = "clusterrolebindings"
+	// ClusterRoles is the clusterroles.rbac.authorization.k8s.io resource
+	ClusterRoles Resource = "clusterrole"
+	// CustomResourceDefinitions is the customresourcedefinitions.apiextensions.k8s.io resource
+	CustomResourceDefinitions Resource = "customresourcedefinitions"
+	// All is the representation of '*' meaning all resources
+	All Resource = "'*'"
+	// Create represents the create verb
+	Create Verb = "create"
+	// Delete represents the delete verb
+	Delete Verb = "delete"
+	// Get represents the get verb
+	Get Verb = "get"
+	// List represents the list verb
+	List Verb = "list"
+	// Update represents the update verb
+	Update Verb = "use"
+	// Watch represents the watch verb
+	Watch Verb = "watch"
+)
+
+// CanI will take a verb and a list of resources and it will check whether the current user / service account can
+// perform that verb against the resources in the Kubernetes cluster
+func CanI(verb Verb, resources ...Resource) (bool, []error) {
+	errList := []error{}
+	for _, resource := range resources {
+		c := exec.Command("kubectl", "auth", "can-i", "--all-namespaces", string(verb), string(resource)) // #nosec
```

Comment:
There is no way to do this with the REST client? 
If we need to use _kubectl_, I think we should use the '-q' option and then just look at the exit status to determine whether an action is allowed. Checking against a string is fragile since there is no guarantee that is won't change and even more it won't work with an internationalized version of `kubectl` (if there is one).

Reply:
Looking at https://kubernetes.io/docs/reference/access-authn-authz/authorization/, it seems that under the hood of `kubectl auth can-i` is the SelfSubjectAccessReview API. For this there are generated client-go functions I believe. One probably could do this via the REST API. WDYT?